tests/unit/client.py

```diff
@@ -248,21 +250,18 @@ def avoid_network_race_condition():
     Used to avoid race condition between subsequent network calls
     on the same object.
     """
-
-    from flask.ext.sqlalchemy import SQLAlchemy
-    from haas.flaskapp import app
-    from haas.model import Node, Project, NetworkingAction
-
     global db_dir
     uri = 'sqlite:///'+db_dir+'/haas.db'
     db = SQLAlchemy(app)
     app.config.update(SQLALCHEMY_DATABASE_URI=uri)
 
     for timeout in range(10):
-        if (db.session.query(NetworkingAction).count() > 0):
+        que = db.session.query(NetworkingAction).count()
+        if (que > 0):
             time.sleep(1)
         else:
-            break
+            return que
+    return timeout
```

Comment:
I would just raise an exception here (of a newly-declared type, which denotes a timeout), and get rid of the if/else checks elsewhere.